### DISCUSSION TITLE
能不能 提出一套 wechat的  http api标准

### DISCUSSION THREAD

**Discussion by yangguangwuwu** [howmanybots/onebot#82] [想法 / 建议]
能不能 提出一套 wechat的  http api标准

**Comment by chenxuan353** [comment]
wechat有什么仓库推荐吗

**Comment by Ink-33** [comment]
> wechat有什么仓库推荐吗

https://github.com/wechaty/wechaty

**Comment by ExerciseBook** [comment]
应该在后续去QQ化的时候接入起来就方便一点了。

**Comment by chenxuan353** [comment]
我知道的情况是在考虑通用兼容标准了，看issue是要等生态活跃一点再弄

**Comment by ExerciseBook** [comment]
现在我感觉主要是缺人编写其他平台的 onebot 实现？（比如说 Discord 平台，Mattermost 平台等）

感觉一个协议的细节，要根据实际情况讨论才可以设计得到。空谈比较难搞。

**Comment by chenxuan353** [comment]
emm，我目前有在写一个，之前cq炸了的时候就在写...不过具体哪些平台比较方便我不太清楚，所以目前还是只做了个兼容层。

**Comment by ExerciseBook** [comment]
如果可以的话，可以加一下咱们的群（ tg / qq 均可）一同交流细节。

**Comment by chenxuan353** [comment]
那么如何加入呢(QQ)

**Comment by ExerciseBook** [comment]
@chenxuan353 啊这，在readme里没群号吗。。。

**Comment by ExerciseBook** [comment]
@chenxuan353 QQ群：201865589
TG群我不知道群号 qwq

**Comment by chenxuan353** [comment]
啊 这个啊

**Comment by ExerciseBook** [comment]
telegram 是 @cqhttp

**Comment by yangguangwuwu** [comment]
兼容每种app的比较难弄 不太现实   还是一种app 一个通用的http api 标准 这个 比较现实 例如 微信 一个 qq一个 tg一个

**Comment by chenxuan353** [comment]
可以做部分范围兼容 然后分bot拓展，至少在群聊私聊临时会话 原路径回复 指定路径回复 这些基础功能上能统一。大部分插件基本上也就是需要接收和推送消息。

**Comment by Ink-33** [comment]
事实上各个聊天平台核心功能无非就是at，回复，图片，文字之类的。可以整几个通用API，各个平台的特色功能自由发挥即可。

**Comment by chenxuan353** [comment]
确实。连接的时候提供一个功能可用性列表就好了。

**Comment by undefined-moe** [comment]
之前不就有 can_send_image 这种东西吗（
用这个方式做 can_recall_msg can_send_video can_edit_msg这种东西不就行了（

**Comment by chenxuan353** [comment]
直接在连接的时候提供bot标识和可用性列表就不用每次都提供了 框架内部直接查功能表就行了。

**Comment by DZ-IO** [comment]
> 之前不就有 can_send_image 这种东西吗（
> 用这个方式做 can_recall_msg can_send_video can_edit_msg这种东西不就行了（

我倒是觉着可以在`get_version_info`里面加入支持的api列表

**Comment by DZ-IO** [comment]
已经[有人](https://github.com/zhamao-robot/wechat-bot)有这个想法了，还有人要做[b站](https://github.com/Metric-Void/bili-onebot)的

**Comment by ExerciseBook** [comment]
@daze456 先插个眼 回头我把 B 站的那个添加到列表。感谢投稿。

**Comment by yangguangwuwu** [comment]
没人实现的主要原因 感觉是 触动了他们的利益

**Comment by LamGC** [comment]
> 直接在连接的时候提供bot标识和可用性列表就不用每次都提供了 框架内部直接查功能表就行了。

然后变成了类似 JavaScript BOM 的兼容性检查（

**Comment by stdrc** [comment]
> 事实上各个聊天平台核心功能无非就是at，回复，图片，文字之类的。可以整几个通用API，各个平台的特色功能自由发挥即可。

是这样的，我设想的 v12 就是划分一些核心 API，然后不同 IM 平台特有的作为扩展，再其他的就是各实现自由发挥（但现在没空整理2333

**Comment by yangguangwuwu** [comment]
统一的标准 写一套 可以共用很多 im 就美滋滋

**Comment by ieew** [comment]
@yangguangwuwu 我感觉有点难，有些bot的消息在设计上就和其他bot不兼容。。
